operator-common/src/test/java/io/strimzi/operator/common/operator/resource/KafkaBridgeCrdOperatorIT.java

```diff
@@ -119,153 +61,32 @@ protected KafkaBridge getResource() {
                 .build();
     }
 
-    @Test
-    public void testUpdateStatus(VertxTestContext context) {
-        Checkpoint async = context.checkpoint();
-
-        log.info("Getting Kubernetes version");
-        PlatformFeaturesAvailability.create(vertx, client)
-            .setHandler(context.succeeding(pfa -> context.verify(() -> {
-                assertThat("Kubernetes version : " + pfa.getKubernetesVersion() + " is too old",
-                        pfa.getKubernetesVersion().compareTo(KubernetesVersion.V1_11), is(not(lessThan(0))));
-            })))
-
-            .compose(pfa -> {
-                log.info("Creating resource");
-                return kafkaBridgeOperator.reconcile(namespace, RESOURCE_NAME, getResource());
-            })
-            .setHandler(context.succeeding())
-
-            .compose(rrCreated -> {
-                KafkaBridge newStatus = new KafkaBridgeBuilder(kafkaBridgeOperator.get(namespace, RESOURCE_NAME))
-                        .withNewStatus()
-                        .withConditions(new ConditionBuilder()
-                                .withType("Ready")
-                                .withStatus("True")
-                                .build())
-                        .endStatus()
-                        .build();
-
-                log.info("Updating resource status");
-                return kafkaBridgeOperator.updateStatusAsync(newStatus);
-            })
-            .setHandler(context.succeeding())
-
-            .compose(rrModified -> kafkaBridgeOperator.getAsync(namespace, RESOURCE_NAME))
-            .setHandler(context.succeeding(modifiedKafkaBridge -> context.verify(() -> {
-                assertThat(modifiedKafkaBridge.getStatus().getConditions().get(0).getType(), is("Ready"));
-                assertThat(modifiedKafkaBridge.getStatus().getConditions().get(0).getStatus(), is("True"));
-            })))
-
-            .compose(rrModified -> {
-                log.info("Deleting resource");
-                return kafkaBridgeOperator.reconcile(namespace, RESOURCE_NAME, null);
-            })
-            .setHandler(context.succeeding(rrDeleted ->  async.flag()));
+    @Override
+    protected CustomResource getResourceWithModifications(CustomResource resourceInCluster) {
```

Comment:
You should be able to `extend` the non-raw type in the class declaration so this method is better typed and avoid having the downcast on the following line